tests/Integration/Behaviour/Features/Scenario/Product/update_prices.feature

```diff
@@ -0,0 +1,141 @@
+#        #todo: rounding issue. #19620
+#        #| unit_price         | 900         |
+        | unity              | bag of ten      |
+        | unit_price_ratio   | 0.222222        |
+
+      Scenario: I update product prices with negative values
+        Given I add product "product2" with following information:
+          | name       | en-US: white hat  |
+          | is_virtual | false             |
+        And I update product "product2" prices with following information:
+          | price           | 50           |
+        And product "product2" has following values:
+          | price           | 50           |
+          | ecotax          | 0            |
+          | wholesale_price | 0            |
+          | unit_price      | 0            |
+        When I update product "product2" prices with following information:
+          | price           | -20          |
+        Then I should get error that product "price" is invalid
+        When I update product "product2" prices with following information:
+          | ecotax          | -2           |
+        Then I should get error that product "ecotax" is invalid
+        When I update product "product2" prices with following information:
+          | wholesale_price | -35          |
+        Then I should get error that product "wholesale price" is invalid
+        When I update product "product2" prices with following information:
+          | unit_price      | -300         |
+        Then I should get error that product "unit price" is invalid
+
+      Scenario: I update product unit price without when product price is 0
+        Given I add product "product3" with following information:
+          | name       | en-US: black hat  |
+          | is_virtual | false             |
+        And product "product3" has following values:
+          | price           | 0            |
+          | unit_price      | 0            |
+        When I update product "product3" prices with following information:
+          | unit_price      | 300          |
+        Then I should get error that product "unit price" is invalid
+
+      Scenario: I update product unit price along with product price
+        Given I add product "product4" with following information:
+          | name       | en-US: blue dress  |
+          | is_virtual | false              |
+        And product "product4" has following values:
+          | price            | 0            |
+          | unit_price       | 0            |
+        When I update product "product4" prices with following information:
+          | price            | 20           |
+          | unit_price       | 500          |
+        Then product "product4" should have following values:
+          | price            | 20           |
+          | unit_price       | 500          |
+          | unit_price_ratio | 0.04         |
+        When I update product "product4" prices with following information:
+          | price            | 0            |
```

Comment:
So the 0 value is purely ignored.. this seems weird not to be able to set 0 if you want to

Reply:
hmm its wrong.
I think price should be reset to 0 in this case along with the unit_price and_unit_price ratio :thinking:

Reply:
edited - the price has priority now, if its 0 then it resets unit_price and ratio to 0 aswell.